### ISSUE TITLE
Inconsistent behaviour

### ISSUE THREAD

**Issue by Natim** [Kinto/kinto-changes#35]
https://firefox.settings.services.mozilla.com/v1/buckets/monitor/collections/changes/records

Contains:

```{
    "data": [
        {
            "id": "0e543556-43bf-3139-1fda-2a0068116c6d",
            "last_modified": 1494453225971,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "certificates",
            "bucket": "blocklists"
        },
        {
            "id": "94626014-57b2-4f7e-90c6-989e00da4269",
            "last_modified": 1494453225971,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "certificates",
            "bucket": "blocklists-preview"
        },
        {
            "id": "3ace9d8e-00b5-a353-7fd5-1f081ff482ba",
            "last_modified": 1494367062674,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "plugins",
            "bucket": "blocklists-preview"
        },
        {
            "id": "c6671b58-a04c-2753-1904-decfb584f9ce",
            "last_modified": 1494024191520,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "addons",
            "bucket": "blocklists-preview"
        },
        {
            "id": "b7f595f9-5fc5-d863-b5dd-e5425dcf427a",
            "last_modified": 1494024191520,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "addons",
            "bucket": "blocklists"
        },
        {
            "id": "19e79f22-62cf-92e1-c12c-a3b4b9cf51be",
            "last_modified": 1491948745529,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "plugins",
            "bucket": "blocklists"
        },
        {
            "id": "deefc6d1-267f-0445-3168-232db4582d75",
            "last_modified": 1485794868067,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "pins",
            "bucket": "pinning"
        },
        {
            "id": "be77064f-af40-d297-93f9-91e676ec04ea",
            "last_modified": 1480349135384,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "gfx",
            "bucket": "blocklists-preview"
        },
        {
            "id": "3fadb169-e5de-a2f0-374e-6eeb3eac3dbb",
            "last_modified": 1480349135384,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "gfx",
            "bucket": "blocklists"
        },
        {
            "id": "061d4584-acd0-fecc-b5fb-dcfe0ad5e123",
            "last_modified": 1475766933820,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "qa",
            "bucket": "blocklists"
        },
        {
            "id": "5acdcc50-306a-b400-c7cb-6812ae023f48",
            "last_modified": 1467983589389,
            "host": "firefox.settings.services.mozilla.com",
            "collection": "certificates,",
            "bucket": "blocklists"
        }
    ]
}
```

- The first inconsistency is that blocklists/certificates is present twice with two distincts last_modified values.

- The second inconsistency is that the last_modified value for blocklists/certificates is different from the collection one:

```
$ http https://firefox.settings.services.mozilla.com/v1/buckets/blocklists/collections/certificates
HTTP/1.1 200 OK
ETag: "1494461909387"

{
    "data": {
        "id": "certificates", 
        "last_modified": 1494461909387
   }
}
```

Or

```
$ http https://firefox.settings.services.mozilla.com/v1/buckets/blocklists/collections/certificates/records
HTTP/1.1 200 OK
ETag: "1494458678750"
```

tldr: 1494453225971 != 1494461909387 != 1494458678750

**Comment by leplatrem** [comment]
> The first inconsistency is that blocklists/certificates is present twice with two distincts last_modified values.

I think it's there only once (the other having a trailing comma).

> tldr: 1494453225971 != 1494461909387 != 1494458678750

It's supposed to give the `collection_timestamp()` of the `record` resource. Did you try on  a local instance (ie. without CDN?)

**Comment by Natim** [comment]
> I think it's there only once (the other having a trailing comma).

Good point :)

**Comment by Natim** [comment]
> Did you try on a local instance (ie. without CDN?)

I have the same results with the local instance.

**Comment by Natim** [comment]
Apparently the bug is not there anymore in production, we couldn't reproduce locally.

**Comment by Natim** [comment]
So we still had the same issue with the blocklists/plugins thing, and it appears to be a Cache Invalidation issue.

```python
>>> from lambda_function import validate_changes_collection
>>> validate_changes_collection({"server": "https://firefox.settings.services.mozilla.com/v1"}, None)
Looking at /buckets/monitor/collections/changes: Etag OK for blocklists/certificates : 1494458678750
Etag OK for blocklists-preview/certificates : 1494458678750
Etag OK for blocklists-preview/plugins : 1494367062674
Etag OK for blocklists-preview/addons : 1494024191520
Etag OK for blocklists/addons : 1494024191520
Etag NOT OK for blocklists/plugins : 1491948745529 != 1494367062674
Etag OK for pinning/pins : 1485794868067
Etag OK for blocklists-preview/gfx : 1480349135384
Etag OK for blocklists/gfx : 1480349135384
Etag OK for blocklists/qa : 1475766933820
```

Without the CDN:
```pycon
Looking at /buckets/monitor/collections/changes: Etag OK for blocklists/certificates : 1494458678750
Etag OK for blocklists-preview/certificates : 1494458678750
Etag OK for blocklists/plugins : 1494367062674
Etag OK for blocklists-preview/plugins : 1494367062674
Etag OK for blocklists-preview/addons : 1494024191520
Etag OK for blocklists/addons : 1494024191520
Etag OK for pinning/pins : 1485794868067
Etag OK for blocklists-preview/gfx : 1480349135384
Etag OK for blocklists/gfx : 1480349135384
Etag OK for blocklists/qa : 1475766933820
```

**Comment by Natim** [comment]
Refs https://github.com/Kinto/kinto-signer/issues/176
